src/globus_sdk/experimental/auth_requirements_error/_validators.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import typing as t
+
+from ._serializable import Serializable
+
+S = t.TypeVar("S", bound=Serializable)
+
+
+class ValidationError(ValueError):
+    pass
+
+
+def str_(name: str, value: t.Any) -> str:
+    if not isinstance(value, str):
+        raise ValidationError(f"'{name}' must be a string")
+    return value
```

Comment:
I recommend ending on `ValidationError`s throughout, like:

```suggestion
    if isinstance(value, str):
        return value
    raise ValidationError(f"'{name}' must be a string")
```

Reply:
Sure thing; will apply momentarily as I catch up on other notes.
(The rest of these I'll just :+1: as I go.)